benchmarks/nemo/nemolite2d/psykal/README.md

```diff
@@ -12,36 +12,63 @@ to set things up for Gnu:
 
     > . ../../../../compiler_setup/gnu.sh
 
+However, if the build target requires the MPI library, the `F90` environment
+variable should be changed to the desired MPI compiler wrappers, for example:
+
+    > export F90=mpif90
+
+Note that the MPI compiler wrapper should call the same compiler family as the
+loaded by the compiler setup script, otherwise the build system compiler flags
+will be unrecognised. It is also advisable to perform a `make allclean` action
+when requesting a new `F90` compiler in order to compile all dependencies with
+the same parameters.
+
 Since the Makefile uses PSyclone you must have PSyclone installed on your
 system (see http://psyclone.readthedocs.io/en/stable/getting_going.html).
 
-The Makefile supports two targets:
+The Makefile supports many targets:
+
+- nemolite2d_serial (default) - vanilla (serial) version
+- nemolite2d_omp - OpenMP version with static scheduling
+- nemolite2d_mpi - MPI version for distributed memory parallelism
+- nemolite2d_hybrid - MPI and OpenMP combined version
+- nemolite2d_acc - OpenACC (Not working - PSyclone will not generate it)
+- nemolite2d_ocl - OpenCL version
+- nemolite2_mpiocl - MPI and OpenCL combined version (Not working - Psyclone
+will generate a OpenCL-only version)
 
-1. nemolite2d_gen - vanilla (serial) version
-2. nemolite2d_gen_omp - OpenMP version with static scheduling
+Executing `make <target>` will generate a folder with the same name of the
+target, containing at least a `alg.f90` and a `psy.f90` for the
+PSyclone-generated Algorithm and PSy-layer components respectively.
+It will also generate kernel source files when necessary. Finally, it will
+build a `nemolite2d.exe` with the source files in that target folder.
 
-but just executing `make` will build both and should result in the
-creation of two executables - `nemolite2d_gen.exe` and
-`nemolite2d_gen_omp.exe`. (The PSyclone-generated PSy code for these
-two binaries can be found in `psy.f90` and `psy_omp.f90`, respectively.)
 
 ## Running ##
 
 Model parameters (size of domain [jpiglo,jpjglo], number of time-steps
 [nitend], whether or not and how often to do output [irecord]) may be
 configured by editing the `namelist` file.
 
-Since `nemolite2d_gen.exe` is serial, it can just be executed on
-the command line, e.g.:
-
-    > ./nemolite2d_gen.exe
-
-For the OpenMP version, you will need to set OMP_NUM_THREADS to the
-desired number of threads to use.
-
-Note that if you are benchmarking performance then you probably need
-to ensure that the threads are bound sensibly to processor cores (see
-e.g. taskset, likwid or Intel's KMP_AFFINITY).
+There are differences on how to execute the generated `nemolite2d_gen.exe`,
```

Comment:
s/on how/in how/

Reply:
Done, also removed the "_gen" in the binary name.